machine-learning/sagemaker-edge-manager/create_components.py

```diff
@@ -0,0 +1,237 @@
+import argparse
+import json
+import os
+import shutil
+
+import boto3
+
+
+def create_artifacts():
+    for component in os.listdir(artifacts_path):
+        component_path = os.path.join(artifacts_path, component)
+        for version in os.listdir(component_path):
+            version_path = os.path.join(component_path, version)
+            build = os.path.join(build_artifacts_path, component, version)
+            os.makedirs(build, mode=0o777, exist_ok=True)
+            for file in os.listdir(version_path):
+                file_path = os.path.join(version_path, file)
+                build_file = os.path.join(build, file)
+                if os.path.isdir(file_path):
+                    shutil.make_archive(build_file, "zip", file_path)
+                    zip_file = "{}.zip".format(build_file)
+                    rel_path = os.path.relpath(zip_file, build_dir_path)
+                    upload_artifacts(zip_file, rel_path)
+                elif os.path.isfile(file_path):
+                    rel_path = os.path.relpath(build_file, build_dir_path)
+                    shutil.copy(file_path, build_file)
+                    upload_artifacts(build_file, rel_path)
+            if "Model" in component:
+                rel_path = os.path.relpath(build, build_dir_path)
+                copy_sample_models(rel_path, inference_models[component])
+
+
+def create_recipes():
+    os.makedirs(build_recipes_path, mode=0o777, exist_ok=True)
+    for component_recipe in os.listdir(recipes_path):
+        with open(
+            os.path.join(recipes_path, component_recipe),
+        ) as f:
+            recipe = json.load(f)
+        component = component_recipe.split("-")
+        c_name = component[0]
+        c_version = component[1].split(".json")[0]
+        recipe["ComponentVersion"] = get_latest_version(c_name, c_version)
+        for manifest in recipe["Manifests"]:
+            for artifact in manifest["Artifacts"]:
+                artifact["Uri"] = artifact["Uri"].replace("$BUCKETNAME$", bucket)
+                artifact["Uri"] = artifact["Uri"].replace("$REGION$", region)
+        recipe_file_name = "{}-{}.json".format(c_name, recipe["ComponentVersion"])
+        with open(os.path.join(build_recipes_path, recipe_file_name), "w") as f:
+            f.write(json.dumps(recipe, indent=4))
+
+
+def get_latest_version(c_name, c_version):
+    try:
+        account = get_account_number()
+        print("Fetching the component {} from the account: {}".format(c_name, account))
+        response = greengrass_client.list_component_versions(
+            arn="arn:aws:greengrass:{}:{}:components:{}".format(region, account, c_name),
+        )
+        component_versions = response["componentVersions"]
+        if len(component_versions) > 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`.

 [Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)